src/network.ts

```diff
@@ -0,0 +1,155 @@
+import * as Bluebird from 'bluebird';
+import * as _ from 'lodash';
+import { fs } from 'mz';
+import * as networkCheck from 'network-checker';
+import * as os from 'os';
+import * as url from 'url';
+
+import * as constants from './lib/constants';
+import { EEXIST } from './lib/errors';
+import { checkTruthy } from './lib/validation';
+
+import blink = require('./lib/blink');
+
+const networkPattern = {
+	blinks: 4,
+	pause: 1000,
+};
+
+let isConnectivityCheckPaused = false;
+let isConnectivityCheckEnabled = true;
+
+function checkHost(
+	opts: networkCheck.ConnectOptions,
+): boolean | PromiseLike<boolean> {
+	if (!isConnectivityCheckEnabled || isConnectivityCheckPaused) {
+		return true;
+	}
+	return networkCheck.checkHost(opts);
+}
+
+function customMonitor(
+	options: networkCheck.ConnectOptions,
+	fn: networkCheck.MonitorChangeFunction,
+) {
+	return networkCheck.monitor(checkHost, options, fn);
+}
+
+export function enableCheck(enable: boolean) {
+	isConnectivityCheckEnabled = enable;
+}
+
+async function vpnStatusInotifyCallback(): Promise<void> {
+	try {
+		await fs.lstat(constants.vpnStatusPath);
+		isConnectivityCheckPaused = true;
+	} catch {
+		isConnectivityCheckPaused = false;
+	}
+}
+
+export const startConnectivityCheck = _.once(
+	async (
+		apiEndpoint: string,
+		enable: boolean,
+		onChangeCallback?: networkCheck.MonitorChangeFunction,
+	) => {
+		enableConnectivityCheck(enable);
+		if (!Boolean(apiEndpoint)) {
+			console.log('No API endpoint specified, skipping connectivity check');
+			return;
+		}
+
+		Bluebird.resolve(fs.mkdir(constants.vpnStatusPath))
+			.catch(EEXIST, () => {
+				console.log('VPN status path exists.');
+			})
+			.then(() => {
+				fs.watch(constants.vpnStatusPath, vpnStatusInotifyCallback);
+			});
+
+		if (enable) {
+			vpnStatusInotifyCallback();
+		}
+
+		const parsedUrl = url.parse(apiEndpoint);
+		const port = parseInt(parsedUrl.port!, 10);
+
+		customMonitor(
+			{
+				host: parsedUrl.hostname,
+				port: port || (parsedUrl.protocol === 'https' ? 443 : 80),
+				path: parsedUrl.path || '',
```

Comment:
This is new?  What's it for?  Also I'd rather review typescript conversions that are purely conversions and don't introduce wholly new code as it's very hard to spot new changes when the entire file is new

Reply:
It was required by the typings.

Reply:
I'm guessing that's because the types are wrong if it was working before?  Does this mean the network check will now be hitting a different path? (it might be relevant depending on what the path is)

Reply:
The path will default to `'/'`, so it will hit the same path, I should probably change the `|| ''` though, I had meant to put `/`.